### ISSUE TITLE
Pull requests without dedicated branch fail creating the matrix

### ISSUE THREAD

**Issue by boesing** [laminas/laminas-ci-matrix-action#24]
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 1.5.2

#### Summary

It seems that for pull requests, which are created from the same branch in the forked repository, it is not possible to create a diff for the matrix generation.

#### Current behavior

Target branch: `1.2.x`
Project branch: `1.2.x`

```
Cloning into '.'...
Cloning repository
Checking out branch 1.2.x
Already on '1.2.x'
Your branch is up to date with 'origin/1.2.x'.
Fetching target ref refs/pull/7/merge
fatal: Refusing to fetch into current branch refs/heads/1.2.x of non-bare repository
```

https://github.com/laminas/laminas-mvc-i18n/pull/7/checks?check_run_id=2254661339

#### How to reproduce

Fork a repository, do not create a dedicated feature/bugfix/whatever branch but push to the same branch, the PR is targeted to later on.

#### Expected behavior

Matrix generation works without conflicting when creating a diff.
Works when the source branch differs: https://github.com/laminas/laminas-mvc-i18n/pull/10/checks?check_run_id=2254711601

**Comment by boesing** [comment]
Uh, interesting fact. I've created a dedicated PR with the same commit but other source branch and the same checks are applied for both PRs:

https://github.com/laminas/laminas-mvc-i18n/pull/7
https://github.com/laminas/laminas-mvc-i18n/pull/10

**Comment by gpgkd906** [comment]
@boesing 
I think I have meet the same problem, with this [PR](https://github.com/laminas/laminas-permissions-rbac/pull/13)
how can i fix this?

**Comment by boesing** [comment]
@gpgkd906 Best practice is to have a dedicated branch for anything you are working on in git.
So you should create a feature branch, e.g. `feature/php80`, and use that for PRs.

You can use this cheat sheet for more details on how to create that branch: https://education.github.com/git-cheat-sheet-education.pdf

**Comment by gpgkd906** [comment]
@boesing 
Thanks very much.
I create a feature branch ```feature/php80 ``` and create a new [PR](https://github.com/laminas/laminas-permissions-rbac/pull/20) then it passed.
the interesting things is, when new PR passed, the old [PR](https://github.com/laminas/laminas-permissions-rbac/pull/13) which was failed, now is passed too.

**Comment by boesing** [comment]
> the interesting things is, when new PR passed, the old [PR](https://github.com/laminas/laminas-permissions-rbac/pull/13) which was failed, now is passed too.

Thats exactly what I've found out in my [second comment in this issue](https://github.com/laminas/laminas-ci-matrix-action/issues/24#issuecomment-812577368). 👍🏼

**Comment by boesing** [comment]
Fixed with [v1.7.1](https://github.com/laminas/laminas-ci-matrix-action/releases/tag/1.7.1)
